src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java

```diff
@@ -2570,6 +2570,18 @@ public void test_submodule_update() throws Exception {
         assertFixSubmoduleUrlsThrows();
     }
 
+    @NotImplementedInJGit
+    public void test_submodule_update_shallow() throws Exception {
```

Comment:
Other test cases would also be useful:
* no `shallow(...)` call
* `shallow(false)`
* `depth(...)`